.github/workflows/integration_tests.yml

```diff
@@ -126,18 +126,29 @@ jobs:
       run: cd tests/playwright && npx playwright test
 
   notify-of-test-run-completion:
-    if: always()
     runs-on: ubuntu-latest
     needs: [notify-of-test-run-start, integration-tests, playwright-tests]
     steps:
       - uses: actions/checkout@v4
-      - name: Notify via Slack of test-run outcome
+      - name: Notify via Slack of test-run success
+        if: needs.integration-tests.result == "success" && needs.playwright-tests.result == "success"
```

Comment:
Is this valid in YAML?

Reply:
if you mean bool operators outside of expression then yes, should be, docs https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-contexts show it inside an expression but unless the if starts with `!` you shouldn't need to enclose it.

Reply:
Note to self: To not having to worry about beginning the value with yaml special characters (or having to use expressions just to escape bool operations in case a negation is needed), the yaml `>` fold block is quite a good hack valid even when the context/property begins with "`!`": _(taken from older gh runner docs)_

```yaml
jobs:
  upload:
    runs-on: ubuntu-latest
    if: >
      github.event.workflow_run.event == 'pull_request' &&
      github.event.workflow_run.conclusion == 'success'

```